src/thor/trace_route_action.cc

```diff
@@ -127,13 +127,16 @@ void thor_worker_t::route_match(Api& request) {
   // TODO - make sure the trace has timestamps..
   auto& options = *request.mutable_options();
   if (RouteMatcher::FormPath(mode_costing, mode, *reader, trace, options, m_path_infos)) {
-    // TODO: we dont support multileg here as it ignores location types
-
-    // TODO: if this were a time dependent match and you werent use_timestamps
-    // you need to propogate the date information to each legs origin location
-    // because triplegbuilder relies on it. form path set the first one but
-    // on the subsequent legs we will need to set them by doing time offsetting
-    // like is done in route_action.cc thor_worker_t::depart_at
+    // TODO: we dont support multileg here as it ignores location types but...
+    // if this were a time dependent match you need to propogate the date time
+    // information to each legs origin location because triplegbuilder relies on it.
+    // form path set the first one but on the subsequent legs we will need to set them
+    // by doing time offsetting like is done in route_action.cc thor_worker_t::depart_at
+
+    // For now we ignore multileg complications and just make sure the searched locations
+    // get the same data information the shape informations had
+    if (options.shape(0).has_date_time())
+      options.mutable_locations(0)->set_date_time(options.shape(0).date_time());
```

Comment:
here because triplegbuilder gets `locations` and not `shape` in the edge walking case, we need to migrate the time information to the `locations` from the `shape`.